model/source_http.go

```diff
@@ -14,7 +25,46 @@ type SourceCreateRequest struct {
 	AppCreationWorkflow string  `json:"app_creation_workflow"`
 	AvailabilityStatus  string  `json:"availability_status"`
 
-	SourceTypeID *int64 `json:"source_type_id"`
+	SourceTypeID    *int64      `json:"-"`
+	SourceTypeIDRaw interface{} `json:"source_type_id"`
+}
+
+// Validate validates that the required fields of the SourceCreateRequest request hold proper values. In the specific
+// case of the UUID, if an empty or nil one is provided, a new random UUID is generated and appended to the request.
+func (req *SourceCreateRequest) Validate() error {
+	if req.Name == nil || *req.Name == "" {
+		return fmt.Errorf("name cannot be empty")
+	}
+
+	// If no valid UUID is provided in the request, we generate one
+	if req.Uid == nil || *req.Uid == "" {
+		id := uuid.New()
+		stringId := id.String()
+		req.Uid = &stringId
+	}
+
+	if !util.SliceContainsString(validWorkflowStatuses, req.AppCreationWorkflow) {
```

Comment:
in ruby version of sources api is also valid when `app_creation_workflow` is not specified - it defaults to `manual_configuration`

Reply:
[You are right](https://github.com/RedHatInsights/sources-api/blob/5befbc497719db35758488535a44afcb5f82000b/db/migrate/20210125173413_add_app_creation_workflow_to_sources.rb#L3). Thank you for pointing this out, I'll default this to `manual_configuration`.

Reply:
[You are right](https://github.com/RedHatInsights/sources-api/blob/5befbc497719db35758488535a44afcb5f82000b/db/migrate/20210125173413_add_app_creation_workflow_to_sources.rb#L3). Thanks for pointing this out, I'll default the parameter to `manual_configuration` if it comes empty / nil.